HW1/P6/P6.py

```diff
@@ -0,0 +1,86 @@
+import time
+import findspark
+findspark.init('/home/zelong/spark')
+import pyspark
+import numpy as np
+
+sc = pyspark.SparkContext()
+
+########################################################################################
+#Clean the data and construct the RDD
+########################################################################################
+def only_capt_letter(x):
+#this function will return true if x contain only letters and all letters are in capital
+	if x.isalpha():
+		if x.isupper():
+			return True
+
+def capt_letter_period(x):
+#return true if string x contains only capitalized letters and end with a period
+	if len(x) > 1:
+		if x[-1] == ".":
+			if x[:-1].isalpha():
+				if x[:-1].isupper():
+					return True	
+
+lines = sc.textFile('pg100.txt')
+words = lines.flatMap(lambda x: x.split(' '))
+words_no_number = words.filter(lambda x: not x.isdigit())
+
+words_no_cap = words_no_number.filter(lambda x: not only_capt_letter(x))
+
+words_filtered = words_no_cap.filter(lambda x: not capt_letter_period(x)).filter(lambda x: not x=='')
+#########################################################################################
+#Shift word RDD
+#Generate RDD with items containing 3 consecutive words
+zip_index = words_filtered.zipWithIndex()
+no_shift = zip_index.map(lambda x: (x[1], ( 0, x[0])))
+left_shift1 = no_shift.map(lambda x: (x[0]-1, (1,x[1][1])))
+left_shift2 = left_shift1.map(lambda x: (x[0] - 1, (2, x[1][1])  ))
+collection_rdd = sc.union([no_shift, left_shift1, left_shift2])
+phrase_rdd = collection_rdd.groupByKey().map(lambda (x,y): (x,[i for i in y])).filter(lambda x: len(x[1]) == 3)
+
```

Comment:
this was a very clever method for getting the triplets of words! i had thought about doing three separate zipWithIndex() calls, but I like how you did it here with a single zipWithIndex() and then creating the left shift, no shift, right shift RDDs and combining them together.